docs/txp.md

```diff
@@ -0,0 +1,434 @@
+
+### Verification
+
+As stated above, we want to verify block `B` validity against GState `S`.
+To do so, each transaction from block's transaction payload is taken separately (in order of inclusion into block) and considered against `S`.
+
+Let's assume that version comparison as described at the end of previos section (Unknown data handling) is done.
+Let's denote flag `verifyAllIsKnown` and assign it value:
+ * `True` if check is to be done
+    - I.e. only known data should be considered valid (unknown data is prohibitied)
+ * `False` otherwise
+
+#### General checks
+
+Checks, that are considering whole transaction, not inputs/outputs in particular.
+
+- *Consistency check*: number of inputs in a transaction must be the same as number of witnesses for this transaction
+- *Attributes known check*: if `verifyAllIsKnown` is `True`, all transaction attributes must be known.
+- *Transaction size check*: size of transaction doesn't exceed size limit
+    - Transaction size is computed as number of bytes in serialized `TxAux` (which contains transaction and its witness)
+    - Transaction size limit is part of `BlockVersionData`.
+    - If the size of transaction is greater than the limit from the adopted `BlockVersionData`, transaction is
+invalid.
+- *Bootstrap era check*: if the block was created during bootstrap era, all
+output addresses must have `BootstrapEraDistr` distribution
+  - Predicate "was created during bootstrap era" is checked with block's slot field
+    and adopted `BlockVersionData` (which contains data which slot is defined last for bootstrap era)
+
+#### Outputs checks
+
+For each output address we perform the following checks:
+1. If `verifyAllIsKnown` is `True`, address' attributes must be known.
+2. If `verifyAllIsKnown` is `True`, address' type must be known.
+3. Address can't be a redeem address.
+
+#### Inputs checks
+
+For each input `i` we perform:
+
+- *Inputs known check*: if `verifyAllIsKnown` is `True`, type of input `i` should be known
+- *UTXO check*: If input `i` is of type `TxInUtxo`, it should have corresponding record in UTXO
+- *Difference check*: input `i` must be different from all other inputs in transaction
+
+In addition we perform number of consolidated checks (checks which consider whole set of inputs):
+
+* Witness check (separately for each witness)
+* Sums check (check inputs sum is greater than or equal to outputs sum)
+* Transaction fee check (check transaction fee is fine)
+
+##### Witness checks
+
+Witness check for any type is two-fold:
+
+* check that witness corresponds to the address of corresponding input
+* check witness is correct
+
+We describe checks for each type in folowing sections.
+
+###### Public key witness
+
+Witness `PkWitness` contains:
+
+* public key `pk`
+* signature `sig`
+
+We require:
+
+* Address is a `PubKey` address
+* Address spending data is created with public key `pk` (\*)
+* Signature `sig` is a valid signature of transaction (validity checked with `pk`)
```

Comment:
**Important**: need an accurate description of how signature check is made:
 
- Which tag used
- What data included into tag

Reply:
Done.